test/smoke/package/src/helpers/TestApplicationSetupManager.ts

```diff
@@ -0,0 +1,337 @@
+        const expoSdkVersion = process.env.EXPO_SDK_MAJOR_VERSION;
+        const pureExpoSdkVersion = process.env.PURE_EXPO_VERSION;
+        this.prepareReactNativeApplication(this.rnWorkspaceDirectory, rnVersion);
+        this.prepareExpoApplication(this.expoWorkspaceDirectory, expoSdkVersion);
+        this.preparePureExpoApplication(this.pureRnWorkspaceDirectory, pureRnVersion, pureExpoSdkVersion);
+        this.prepareHermesApplication(this.hermesWorkspaceDirectory, rnVersion);
+    }
+
+    private static async getLatestSupportedRNVersionForExpo(expoSdkMajorVersion?: string): Promise<any> {
+        const printSpecifiedMajorVersion = expoSdkMajorVersion ? `sdk-${expoSdkMajorVersion}` : "";
+        const printIsLatest = printSpecifiedMajorVersion ? "" : "latest ";
+        console.log(`*** Getting latest React Native version supported by ${printIsLatest}Expo ${printSpecifiedMajorVersion}...`);
+        return new Promise((resolve, reject) => {
+            utilities.getContents("https://exp.host/--/api/v2/versions", null, null, function (error, versionsContent) {
+                if (error) {
+                    reject(error);
+                }
+                try {
+                   const content = JSON.parse(versionsContent);
+                   if (content.sdkVersions) {
+                       let usesSdkVersion: string | undefined;
+                       if (expoSdkMajorVersion) {
+                            usesSdkVersion = Object.keys(content.sdkVersions).find((version) => semver.major(version) === parseInt(expoSdkMajorVersion));
+                            if (!usesSdkVersion) {
+                                console.log(`*** Сould not find the version of Expo sdk matching the specified version - ${printSpecifiedMajorVersion}`);
+                            }
+                       }
+                       if (!usesSdkVersion) {
+                            usesSdkVersion = Object.keys(content.sdkVersions).sort((ver1, ver2) => {
+                                if (semver.lt(ver1, ver2)) {
+                                    return 1;
+                                } else if (semver.gt(ver1, ver2)) {
+                                    return -1;
+                                }
+                                return 0;
+                            })[0];
+                       }
+                       if (content.sdkVersions[usesSdkVersion]) {
+                        if (content.sdkVersions[usesSdkVersion].facebookReactNativeVersion) {
+                            console.log(`*** Latest React Native version supported by Expo ${printSpecifiedMajorVersion}: ${content.sdkVersions[usesSdkVersion].facebookReactNativeVersion}`);
+                            resolve(content.sdkVersions[usesSdkVersion].facebookReactNativeVersion as string);
+                        }
+                    }
```

Comment:
Please pay attention on files formatting. For example you can run `Format Document` command in the document.